cli_widgets/spin_loader.hpp

```diff
@@ -146,6 +146,11 @@ namespace cli_widgets {
       } else {
         _out << termcolor::green;
       }
+
+      // clear the line as it was prior to writing the done message
+      _out << "\r\e[K" << std::flush;
```

Comment:
I'm unsure that this is better. IMHO it can break an FTXUI or another lib drawing something somewhere else later on the same line. 

The idea behind how it was implemented was to treat the console as a pixel rendering space. So that we don't modify where we aren't doing anything.

By the way are these sequences supported on windows cmd too ? I'm unsure of how much actually works cross plaform.

Reply:
it doesn't break the windows console (tested that) the advantage being that the full line is being cleared.

Regarding the "what if we use another lib" part: the spin_loader would break it anyway is so many interesting ways, so meh

Reply:
But that's not true, that's the reason why you could do the use of your cycler. It wouldn't break it, it only writes where it's called.

Reply:
If the widgets starts to behave as such then you cannot write behind or before it.

Reply:
yeah, but that change we're talking about is in the spin_loader which "takes the console" anyway

Reply:
I know it's in the spin_loader and I feel it's a regression regarding supporting of composition on the same line with other printed chars on the same line. Without adding any extra feature / improvements.

I mean why can you write behind a spin_loader, because it doesn't reset the whole line on each redraw, hence why should we suddenly delete the full line when done ?

Reply:
convinced

Reply:
Thanks :-)